test/models/test_client.dart

```diff
@@ -131,8 +127,4 @@ class TestMeiliSearchClient extends MeiliSearchClient {
           ),
     );
   }
-
-  Future<void> _deleteTasksForDeletedIndexes(List<String> indexes) async {
```

Comment:
@ahmednfwela, this code was causing some intermittent trouble in the CI. You can see some examples here:

- https://github.com/meilisearch/meilisearch/actions/runs/7265526184/job/19822841536
- https://github.com/meilisearch/meilisearch-dart/actions/runs/7276040412
- https://github.com/meilisearch/meilisearch-dart/actions/runs/7275849120
- https://github.com/meilisearch/meilisearch-dart/actions/runs/7275802328
- https://github.com/meilisearch/meilisearch-dart/actions/runs/7275359927

Although keeping track of used resources is very clever, it isn't worth it for two reasons. First, it is hard to understand where the error comes from (due to the stack trace dart left behind). And second, because you should always clean up the state before you run the tests, not after. 

I like this article that highlights why this should be followed as a good practice: https://docs.cypress.io/guides/references/best-practices#Using-after-Or-afterEach-Hooks I've been applying this concept for a long time, and it helps a lot.